src/Operators/finitedifference.jl

```diff
@@ -1459,6 +1459,139 @@ end
+- a face-valued vector field `v`
+- a center-valued field `x`
+```math
+U(v,x)[i] = \\begin{cases}
+  v[i] \\left(-2 x[i-\\tfrac{3}{2}] + 10 x[i-\\tfrac{1}{2}] + 4 x[i+\\tfrac{1}{2}] \\right) / 12  \\textrm{, if } v[i] > 0 \\\\
+  v[i] \\left(4 x[i-\\tfrac{1}{2}] + 10 x[i+\\tfrac{1}{2}] -2 x[i+\\tfrac{3}{2}]  \\right) / 12  \\textrm{, if } v[i] < 0
+  \\end{cases}
+```
+This stencil is based on [BorisBook1973](@cite).
+
+Supported boundary conditions are:
+- [`ThirdOrderOneSided(x₀)`](@ref): uses the third-order downwind reconstruction to compute `x` on the left boundary,
+and the third-order upwind reconstruction to compute `x` on the right boundary.
+
+!!! note
+    Similar to the [`Upwind3rdOrderBiasedProductC2F`](@ref) operator, these boundary conditions do not define the value at the actual boundary faces, and so this operator cannot be materialized directly: it needs to be composed with another operator that does not make use of this value, e.g. a [`DivergenceF2C`](@ref) operator, with a [`SetValue`](@ref) boundary.
+"""
+struct FCTBorisBook{BCS} <: AdvectionOperator
+    bcs::BCS
+end
+FCTBorisBook(; kwargs...) = FCTBorisBook(NamedTuple(kwargs))
+
+return_eltype(::FCTBorisBook, V, A) =
+    Geometry.Contravariant3Vector{eltype(eltype(V))}
+
+return_space(
+    ::FCTBorisBook,
+    velocity_space::Spaces.FaceFiniteDifferenceSpace,
+    arg_space::Spaces.CenterFiniteDifferenceSpace,
+) = velocity_space
+return_space(
+    ::FCTBorisBook,
+    velocity_space::Spaces.FaceExtrudedFiniteDifferenceSpace,
+    arg_space::Spaces.CenterExtrudedFiniteDifferenceSpace,
+) = velocity_space
+
+@inline function fct_boris_book(v, a⁻, a⁺, a⁺⁺, a⁺⁺⁺, step)
+    if v != zero(eltype(a⁻))
+        sign(v) ⊠ (RecursiveApply.rmap(
+            max,
+            zero(eltype(a⁻)),
+            RecursiveApply.rmap(
+                min,
+                RecursiveApply.rmap(abs, v),
+                RecursiveApply.rmap(
+                    min,
+                    sign(v) ⊠ (a⁺⁺⁺ - a⁺⁺) ⊠ step,
+                    sign(v) ⊠ (a⁺ - a⁻) ⊠ step,
+                ),
+            ),
+        ))
+    else
+        RecursiveApply.rmap(
+            max,
+            zero(eltype(a⁻)),
+            RecursiveApply.rmap(
+                min,
+                v,
+                RecursiveApply.rmap(min, (a⁺⁺⁺ - a⁺⁺) ⊠ step, (a⁺ - a⁻) ⊠ step),
+            ),
+        )
+    end
+end
+
+stencil_interior_width(::FCTBorisBook, velocity, arg) =
+    ((0, 0), (-half, half + 2))
+
+@inline function stencil_interior(::FCTBorisBook, loc, idx, hidx, velocity, arg)
+    space = axes(arg)
+    a⁻ = getidx(arg, loc, idx - half, hidx)
+    a⁺ = getidx(arg, loc, idx + half, hidx)
+    a⁺⁺ = getidx(arg, loc, idx + half + 1, hidx)
+    a⁺⁺⁺ = getidx(arg, loc, idx + half + 2, hidx)
+    vᶠ = Geometry.contravariant3(
+        getidx(velocity, loc, idx, hidx),
+        Geometry.LocalGeometry(space, idx, hidx),
+    )
+    step = Geometry.LocalGeometry(space, idx, hidx).∂x∂ξ[1]
```

Comment:
`∂x∂ξ[1]` is the only part of this function that I'm not certain about in terms of generality. We may need to dig into how to make sure that we're grabbing `dz` for all space types.

Reply:
I thought about this and as you pointed out yesterday in our chat, for any space, `LocalGeometry(space, idx, hidx)` would return the info for all space types. And we use this only for vertical spaces (i.e., vertical center space or vertical face space). I checked that the syntax and indexing is the same in both cases